File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var findMatrix = function(nums) {
    const map = {}
    for (const n of nums) {
        map[n] = map[n] + 1 || 1
    }
    const res = []
    while (Object.values(map).some(n => n > 0)) {
        const subarr = []
        for (const n in map) {
            if (map[n] > 0) {
                map[n]--
                subarr.push(n)
            }
        }
        res.push(subarr)
    }
    return res
};

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        counter = Counter(nums)
        res = [[]]
        for num, count in counter.items():
            cnt = count
            for arr in res:
                if cnt:
                    arr.append(num)
                    cnt -= 1
                else:
                    break
            for _ in range(0, cnt):
                res.append([num])
        return res        
        